src/charm.py

```diff
@@ -54,6 +53,7 @@ def __init__(self, framework: ops.Framework):
         self.files_to_push = []
 
         try:
+            self._validate_principals_config()
             self._validate_repository_config()
         except ErrorWithStatus as e:
             self.unit.status = e.status
```

Comment:
In full honesty, this is an operation that can be done in a component. All other components would depend on this one, so it'd be somewhat similar to the leadership component.

I understand this structure was introduced in a different PR, but I think it's worth mentioning. We can file an issue for this, no further change required in this PR.

Reply:
Yeh, agree. Added a tracking issue
https://github.com/canonical/github-profiles-automator/issues/44